vioscsi/helper.c

```diff
@@ -82,53 +86,55 @@ ENTER_FN_SRB();
             }
         } else {
             RhelDbgPrint(TRACE_LEVEL_ERROR, " StorPortGetStartIoPerfParams failed srb 0x%p status 0x%x MessageNumber %d.\n", Srb, status, param.MessageNumber);
+            // FIXME
+            // Should we return on this error..?
+            // Should it be TRACE_LEVEL_FATAL..?
+            //is_done = TRUE;
         }
     }
-
+    
     srbExt = SRB_EXTENSION(Srb);
-
+    
     if (!srbExt) {
         RhelDbgPrint(TRACE_LEVEL_INFORMATION, " No SRB Extenstion for SRB 0x%p \n", Srb);
+EXIT_FN_SRB();
         return;
     }
-
+    
     MessageID = QUEUE_TO_MESSAGE(QueueNumber);
     index = QueueNumber - VIRTIO_SCSI_REQUEST_QUEUE_0;
-
+    
     if (adaptExt->reset_in_progress) {
-        RhelDbgPrint(TRACE_LEVEL_FATAL, " Reset is in progress, completing SRB 0x%p with SRB_STATUS_BUS_RESET.\n", Srb);
+        RhelDbgPrint(TRACE_LEVEL_WARNING, " Reset is in progress, completing SRB 0x%p with SRB_STATUS_BUS_RESET.\n", Srb);
         SRB_SET_SRB_STATUS(Srb, SRB_STATUS_BUS_RESET);
         CompleteRequest(DeviceExtension, Srb);
+EXIT_FN_SRB();
         return;
     }
-
-    VioScsiVQLock(DeviceExtension, MessageID, &LockHandle, FALSE);
+    VioScsiLockManager(DeviceExtension, MessageID, &LockHandle, FALSE, VIOSCSI_VQLOCKOP_LOCK);
```

Comment:
We have the technology. <-- I hope you get the reference...

Reply:
Just to be clear, the idea here was to permit WPP tracing of each lock/unlock function whilst optimising. The problem with optimising and WPP is that `[%!FUNC!]` becomes the parent function instead. So it makes sense to me that this would form part of the WPP PR. That being said, I can certainly do a separate commit and PR for it.